### PR TITLE
[BD-5] [TNL-7337] Make peer dependent steps skippable 

### PR DESCRIPTION
**Pull request by shimulch**
This PR enables skipping Peer assessment steps (if there is no submission to assess) for students when submitting an ORA response. This unblocks them to assess their own response beforehand without waiting for ``must_grade`` or ``must_be_graded_by`` to satisfy.

**JIRA tickets**: 
https://openedx.atlassian.net/browse/TNL-7337
https://tasks.opencraft.com/browse/BB-2732

**Dependencies**: None

**Screenshots**:
screenshot

**Sandbox URL**: 
- https://bb2732.opencraft.hosting/
- https://studio.bb2732.opencraft.hosting/

**Merge deadline**: None

**Testing instructions**:

1. Create a new open assessment block with Must grade & Grade By equals to ``4``
2. Use an enrolled user to do a submission
3. Complete ``Your Response`` & ``Learn to assess response``
4. ``Assess Peers`` & ``Assess Your Response`` both become in progress, making ``Assess Peers`` skippable 

**Author notes and concerns**:

1. A new function ``can_be_skipped`` introduced in peer API 
2. ``AssessmentWorkflowStep`` model now supports checking if a step is skippable using the ``can_be_skipped`` function in sub-API.
3.  ``AssessmentWorkflowStep`` model has a new boolean field ``skipped`` to track if the step has been skipped.
4. When updating state in ``update_from_assessments`` the new logic is the following - 
Find all unskippable steps that yet to be completed
Find all skippable steps that yet to be completed
If there are no unskippable steps then pop 1st skippable step and use that as an unskippable step
Mark current step as skipped if skippable
Select the next unskippable step and make it the current step

**Reviewers**
- [ ] @0x29a 
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @shimulch! I've created [OSPR-4945](https://openedx.atlassian.net/browse/OSPR-4945) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@shimulch Thank you for your contribution. Please let me know once it is ready for our review.

File: openassessment/assessment/api/peer.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 PEER_TYPE = "PE"
 
 
-def can_be_skipped(submission_uuid, peer_requirements):
+def can_be_skipped(submission_uuid, peer_requirements):  # pylint: disable=unused-argument
```
Comment (giovannicimolin) [inline openassessment/assessment/api/peer.py:27]: Why is this? Can you remove the argument if it's not used?

Comment (shimulch) [reply openassessment/assessment/api/peer.py:27]: I've just tried to keep consistency on API calls. I am calling it from [here](https://github.com/open-craft/edx-ora2/blob/shimulch/bb-2732-make-peer-dependent-steps-skipable/openassessment/workflow/models.py#L844). I kept same functiontion signature as seen as [here](https://github.com/open-craft/edx-ora2/blob/shimulch/bb-2732-make-peer-dependent-steps-skipable/openassessment/workflow/models.py#L913) and [here](https://github.com/open-craft/edx-ora2/blob/shimulch/bb-2732-make-peer-dependent-steps-skipable/openassessment/workflow/models.py#L918).

Previously I was using those arguments to check if there any peer assessment exists. But as now we want it to be always skipped, the argument ``submission_uuid`` is not being used anymore.

Comment (giovannicimolin) [reply openassessment/assessment/api/peer.py:27]: Makes sense, you can keep it in this case. Sorry for the false alarm.

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested that peer assessment step is non-blocking under different conditions.
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~ No new interfaces introduced.
- [x] Includes documentation
- [ ] ~~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~~

**Comment by giovannicimolin** [comment]
@marcotuts You can test out this change in the sandbox here: https://bb2732.opencraft.hosting/courses/course-v1:OpenCraft+test+2020/courseware/c2fbfb71d56340e1a92c00fc3c9739e5/41715896ebba4e1d81221f64113eac19/?activate_block_id=block-v1%3AOpenCraft%2Btest%2B2020%2Btype%40sequential%2Bblock%4041715896ebba4e1d81221f64113eac19

You can use `staff` default account to modify the course and test other scenarios too.

CC @natabene This is ready for a product review.

**Comment by natabene** [comment]
@marcotuts Can you give this a look when you have a chance?

**Review by jnlapierre — APPROVED** [review]
LGTM, merge conflicts will need to be resolved.

**Comment by shimulch** [comment]
@jnlapierre, Merge conflicts resolved and commits are squashed.

**Pull request merged**